src/test/java/dev/openfeature/sdk/providers/memory/InMemoryProviderTest.java

```diff
@@ -105,4 +114,42 @@ void shouldThrowIfNotInitialized() {
         // ErrorCode.PROVIDER_NOT_READY should be returned when evaluated via the client
         assertThrows(ProviderNotReadyError.class, ()-> inMemoryProvider.getBooleanEvaluation("fail_not_initialized", false, new ImmutableContext()));
     }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    void emitChangedFlagsOnlyIfThereAreChangedFlags() {
+        Consumer<EventDetails> handler = mock(Consumer.class);
+        Map<String, Flag<?>> flags = buildFlags();
+
+        OpenFeatureAPI.getInstance().onProviderConfigurationChanged(handler);
+        OpenFeatureAPI.getInstance().setProviderAndWait(provider);
+
+        provider.updateFlags(flags);
+
+        verify(handler, times(1))
+                .accept(argThat(details -> details.getFlagsChanged().isEmpty()));
+    }
+
+    @Test
+    void multithreadedTest() throws InterruptedException {
+        ExecutorService executorService = Executors.newFixedThreadPool(100);
+        List<Callable<Void>> updates = new ArrayList<>();
+        for (int i = 0; i < 10000; i++) {
+            String flagKey = "multithreadedFlag" + i;
+            updates.add(() -> {
+                provider.updateFlag(flagKey, Flag.builder()
+                        .variant("on", true)
+                        .variant("off", false)
+                        .defaultVariant("on")
+                        .build());
+                return null;
+            });
+        }
+
+        executorService.invokeAll(updates);
+
+        for (int i = 0; i < 10000; i++) {
+            assertTrue(client.getBooleanValue("multithreadedFlag" + i, false));
+        }
+    }
```

Comment:
I'm a bit dubious of the value of this test. Did you actually see it fail at least sometimes without your changes?

Reply:
I did write this test before making any changes and did see it had failed sometimes.

After my change, I have not observed any failures in this test.